Dockerfile

```diff
@@ -35,13 +36,34 @@ RUN apk add --no-cache gcc musl-dev && \
 FROM alpine:latest
 LABEL maintainer="Liang Ding<845765@qq.com>"
 
+# Set default values in case PUID/PGID are not provided
+ARG USER_ID=1000
+ARG GROUP_ID=1000
+
+ENV PUID $USER_ID
+ENV PGID $GROUP_ID
+
 WORKDIR /opt/siyuan/
 COPY --from=GO_BUILD /opt/siyuan/ /opt/siyuan/
-RUN addgroup --gid 1000 siyuan && adduser --uid 1000 --ingroup siyuan --disabled-password siyuan && apk add --no-cache ca-certificates tzdata && chown -R siyuan:siyuan /opt/siyuan/
+
+# Create User and Group
+RUN if ! getent group ${PGID}; then \
+      addgroup --gid ${PGID} siyuan; \
+    else \
+      groupname=$(getent group ${PGID} | cut -d: -f1); \
+      echo "Group with GID ${PGID} already exists, using group: $groupname"; \
+    fi && \
+    adduser --uid ${PUID} --ingroup ${groupname:-siyuan} --disabled-password siyuan && \
+    chown -R siyuan:${groupname:-siyuan} /opt/siyuan/
+
+RUN apk add --no-cache ca-certificates tzdata
 
 ENV TZ=Asia/Shanghai
+ENV HOME=/home/siyuan
 ENV RUN_IN_CONTAINER=true
 EXPOSE 6806
+VOLUME /home/siyuan/workspace
```

Comment:
I'm not sure if this will affect users who have specified the `--workspace` parameter, such as the usage described in the README:

```
docker run -d -v workspace_dir_host:workspace_dir_container -p 6806:6806 b3log/siyuan --workspace=workspace_dir_container --accessAuthCode=xxx
```

Reply:
> I'm not sure if this will affect users who have specified the `--workspace` parameter, such as the usage described in the README:
> 
> ```
> docker run -d -v workspace_dir_host:workspace_dir_container -p 6806:6806 b3log/siyuan --workspace=workspace_dir_container --accessAuthCode=xxx
> ```

That's a very valid point. I think I can safely remove this line though, I will test if it impacts the result. 

> Another question, does it work just with docker (not compose)?

Yes, customizing `USER_ID` and `GROUP_ID` is possible with just Docker, not only via Docker Compose. You can pass `--build-arg` during the `docker build` process to specify these values. 

```bash
docker build --build-arg USER_ID=1005 --build-arg GROUP_ID=100 -t siyuan_note_image .
docker run -d --name siyuan_note siyuan_note_image
```

Reply:
I believe that the ownership of the workspace should also change if the workspace is specified.